src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -248,29 +248,58 @@ public IObservable<Unit> UnmarkLocalBranch(ILocalRepositoryModel repository)
             });
         }
 
-        public IObservable<string> ExtractFile(ILocalRepositoryModel repository, string commitSha, string fileName)
+        public IObservable<string> ExtractFile(
+            ILocalRepositoryModel repository,
+            IModelService modelService,
+            string commitSha,
+            string fileName,
+            string fileSha)
         {
             return Observable.Defer(async () =>
             {
                 var repo = gitService.GetRepository(repository.LocalPath);
                 await gitClient.Fetch(repo, "origin");
-                var result = await gitClient.ExtractFile(repo, commitSha, fileName);
+                var result = await gitClient.ExtractFile(repo, commitSha, fileName) ??
+                             await modelService.GetFileContents(repository, commitSha, fileName, fileSha);
                 return Observable.Return(result);
             });
         }
 
-        public IObservable<Tuple<string, string>> ExtractDiffFiles(ILocalRepositoryModel repository, IPullRequestModel pullRequest, string fileName)
+        public IObservable<Tuple<string, string>> ExtractDiffFiles(
+            ILocalRepositoryModel repository,
+            IModelService modelService,
+            IPullRequestModel pullRequest,
+            string fileName,
+            string fileSha)
         {
             return Observable.Defer(async () =>
             {
                 var repo = gitService.GetRepository(repository.LocalPath);
                 await gitClient.Fetch(repo, "origin");
+
+                // The left file is the target of the PR so this should already be fetched.
                 var left = await gitClient.ExtractFile(repo, pullRequest.Base.Sha, fileName);
-                var right = await gitClient.ExtractFile(repo, pullRequest.Head.Sha, fileName);
+
+                // The right file - if it comes from a fork - may not be fetched so fall back to
+                // getting the file contents from the model service.
+                var right = await gitClient.ExtractFile(repo, pullRequest.Head.Sha, fileName) ??
+                            await modelService.GetFileContents(repository, pullRequest.Head.Sha, fileName, fileSha);
                 return Observable.Return(Tuple.Create(left, right));
             });
         }
 
+        async Task<string> ExtractFile(
+            ILocalRepositoryModel repository,
+            IRepository repo,
+            IModelService modelService,
+            string commitSha,
+            string fileName,
+            string fileSha)
+        {
+            return await gitClient.ExtractFile(repo, commitSha, fileName) ??
+                   await modelService.GetFileContents(repository, commitSha, fileName, fileSha);
+        }
```

Comment:
Is this method used somewhere? This seems to be a duplicate of the one above, did you mean to call this one instead?